regulation-worker/cmd/main_test.go

```diff
@@ -42,24 +42,26 @@ func handler() http.Handler {
 func run(m *testing.M) int {
 	svr := httptest.NewServer(handler())
 	defer svr.Close()
-	workspaceID := "1zzAn8ZshcdkLN5TvP86VqLMT90"
+	workspaceID := "216Co97d9So9TkqphM0cxBzRxc3"
 	svcCtx, svcCancel := context.WithCancel(context.Background())
-
+	code := make(chan int)
 	go func() {
-		_ = os.Setenv("workspaceID", workspaceID)
-		_ = os.Setenv("urlPrefix", svr.URL)
-		main.Run(svcCtx)
-		<-c
+
+		os.Setenv("CONFIG_BACKEND_URL", "https://api.dev.rudderlabs.com")
+		os.Setenv("WORKSPACE_TOKEN", "216Co97d9So9TkqphM0cxBzRxc3")
+		os.Setenv("CONFIG_PATH", "./test_config.yaml")
+		config.Load()
+		logger.Init()
+		backendconfig.Init()
+		code <- m.Run()
 		svcCancel()
+
 	}()
-	os.Setenv("CONFIG_BACKEND_URL", "https://api.dev.rudderlabs.com")
-	os.Setenv("WORKSPACE_TOKEN", "1zzAnCvbknUuGogjQIBhkST0O4K")
-	os.Setenv("CONFIG_PATH", "./test_config.yaml")
-	config.Load()
-	backendconfig.Init()
-	code := m.Run()
-
-	return code
+	_ = os.Setenv("workspaceID", workspaceID)
+	_ = os.Setenv("urlPrefix", svr.URL)
+	main.Run(svcCtx)
+	statusCode := <-code
+	return statusCode
```

Comment:
```suggestion
```suggestion
	code := main.Run(svcCtx)
	<-done
	return code
``````